src/frontend/components/Chat/SharedChatComponents/InputDisplayNameOverlay/index.tsx

```diff
@@ -0,0 +1,172 @@
+import { defineMessages, useIntl } from 'react-intl';
+
+import { InputBar } from 'components/Chat/SharedChatComponents/InputBar';
+import { ExitCrossSVG } from 'components/SVGIcons/ExitCrossSVG';
+import { QuestionMarkSVG } from 'components/SVGIcons/QuestionMarkSVG';
+import { converse } from 'utils/window';
+import { InputDisplayNameIncorrectAlert } from './InputDisplayNameIncorrectAlert/index';
+
+const messages = defineMessages({
+  closeButtonTitle: {
+    defaultMessage: 'Click this button to close the overlay.',
+    description: 'A title describing the close button action',
+    id: 'components.InputDiplayNameOverlay.closeButtonTitle',
+  },
+  inputAnonymousKeywordForbiddenAlertMessage: {
+    defaultMessage: 'Keyword "anonymous" is not allowed.',
+    description:
+      'An alert message explaining why the entered display name is invalid.',
+    id: 'components.InputDisplayNameOverlay.inputAnonymousKeywordForbiddenAlertMessage',
+  },
+  inputDisplayNameInformative: {
+    defaultMessage:
+      "The display name is the pseudonym you will be authenticated with on this live. You won't be able to change it during the live. Other participants will see you with this name. The instructor will however be able to see your genuine identity if you have previously identified yourself with the LTI.",
+    description:
+      'An informative text about the display name which is asked to enter.',
+    id: 'components.InputDisplayNameOverlay.inputDisplayNameInformative',
+  },
+  inputDisplayNameLabel: {
+    defaultMessage: 'Display name',
+    description: 'An label describing the input below.',
+    id: 'components.InputDisplayNameOverlay.inputDisplayNameLabel',
+  },
+  inputDisplayNamePlaceholder: {
+    defaultMessage: 'Enter your desired display name...',
+    description: 'The input bar to fill your display name.',
+    id: 'components.InputBar.inputDisplayNamePlaceholder',
+  },
+  inputTooShortAlertMessage: {
+    defaultMessage: 'Min length is 3 characters.',
+    description:
+      'An alert message explaining why the entered display name is invalid.',
+    id: 'components.InputDisplayNameOverlay.inputTooShortAlertMessage',
+  },
+  inputTooLongAlertMessage: {
+    defaultMessage: 'Max length is 15 characters.',
+    description:
+      'An alert message explaining why the entered display name is invalid.',
+    id: 'components.InputDisplayNameOverlay.inputTooLongAlertMessage',
+  },
+});
+
+interface InputDisplayNameOverlayProps {
+  setInputBarActive: React.Dispatch<React.SetStateAction<boolean>>;
+  setOverlay: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const InputDisplayNameOverlay = ({
+  setInputBarActive,
+  setOverlay,
+}: InputDisplayNameOverlayProps) => {
+  const intl = useIntl();
+  const [alertsState, setAlertsState] = useState<string[]>([]);
+
+  const processDisplayName = (displayName: string) => {
+    const alerts: string[] = [];
+    if (displayName.toLowerCase().includes('anonymous')) {
```

Comment:
you should factorise this value (`anonymous`) with the secret id generation function

Reply:
done